WalletConnect v2

obernardovieira opened the pull request:
This PR change utils integration to work with WalletConnect v2 with `@web3modal/react` and `wagmi`. These are two required dependencies. wagmi is in some places used within utils.

In terms of hooks, nothing has changed, only the provider has changed.

The README was updated and now has some examples.

0xarthurxyz commented:
@obernardovieira Would love to know when you merge this PR and hear about your experience using web3modal + wagmi!

obernardovieira commented:
> @obernardovieira Would love to know when you merge this PR and hear about your experience using web3modal + wagmi!

Hi. Sorry, I've missed this before. Even though it's still in PR, we updated the website last week and updated the dapp today. So, right now we are using web3modal and wagmi everywhere. The only reason this is still in PR is because we were still testing a few minor things and this is also used on the wallet and there we didn't finish the integration yet. But that doesn't affect anything. It was super easy. I sincerely recommend anyone to use it.
